File: Python Code/Daily_Flash_Code_0/Week 5/13 Feb/27-DailyFlash_Solutions/27-DailyFlash_Solutions/13_Feb_Solution_Three/Python/prog2.py
h = input("Hex : ")
l = len(h)
for i in h:
    if(not((i >= '0' and i <= '9') or (i >= 'A' and i <= 'F'))):
        print("Enter correct hex")
        exit(1)
i = 0
d = 0
while(l > 0):
    if(h[i] <= '9'):
        num = ord(h[i]) - 48
    else:
        num = ord(h[i]) - 55
    d = d * 16 + num
    i += 1
    l -= 1
print("dec : ", d)

File: Python Code/Daily_Flash_Code_0/Week 5/13 Feb/27-DailyFlash_Solutions/27-DailyFlash_Solutions/13_Feb_Solution_Three/C/prog2.c
#include<stdio.h>
#include<string.h>
#include<stdlib.h>
int main(void) {
	char hex[10];
	printf("Enter hex : ");
	fgets(hex, sizeof(hex), stdin);
	int len = strlen(hex);
	if(hex[len - 1] == '\n') {
		hex[len - 1] = '\0';
	}
	len = strlen(hex);
	int k = 0;
	while(k < len) {
		if(!((hex[k] >= '0' && hex[k] <= '9') || (hex[k] >= 'A' && hex[k] <= 'F'))) {
			printf("Enter valid hex");
			exit(1);
		}
		k++;
	}
	int d = 0, num, i = 0;
	while(len > 0) {
		num = hex[i] - 48 <= 9 ? hex[i] - 48 : hex[i] - 55;
		d = d * 16 + num;
		len--;
		i++;
	}
	printf("Dec = %d", d);
	return 0;
}
